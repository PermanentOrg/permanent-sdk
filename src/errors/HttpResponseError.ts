export class HttpResponseError extends Error {
  public readonly statusCode: number;

  public constructor(statusCode: number, message: string) {
    super(message);
    this.statusCode = statusCode;
  }
}
