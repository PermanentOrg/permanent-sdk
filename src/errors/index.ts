export * from './HttpResponseError';
export * from './ValidationError';
