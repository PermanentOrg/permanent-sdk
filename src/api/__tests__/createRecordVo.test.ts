import path from "node:path";
import nock from "nock";
import { createRecordVo } from "..";
import { ValidationError } from "../../errors";

describe("createRecordVo", () => {
	it("should create a record", async () => {
		nock("https://permanent.local")
			.post("/api/record/registerRecord", {
				s3url: "exampleUrl",
				displayName: "testFile",
				parentFolderId: 1,
				uploadFileName: "testFile.txt",
				fileType: "text/plain",
				size: 12,
				failOnDuplicateName: false,
			})
			.replyWithFile(
				200,
				path.join(__dirname, "fixtures", "createRecordVo", "recordVo.json"),
				{
					"Content-Type": "application/json",
				},
			);

		const recordVo = await createRecordVo(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			{
				s3Url: "exampleUrl",
				displayName: "testFile",
				parentFolderId: 1,
				uploadFileName: "testFile.txt",
				fileType: "text/plain",
				size: 12,
			},
		);

		expect(recordVo).toMatchSnapshot();
	});

	it("should error when provided an invalid response", async () => {
		nock("https://permanent.local")
			.post("/api/record/registerRecord", {
				s3url: "exampleUrl",
				displayName: "testFile",
				parentFolderId: 1,
				uploadFileName: "testFile.txt",
				fileType: "text/plain",
				size: 12,
				failOnDuplicateName: false,
			})
			.reply(200, '["Not at all a s3UploadVo"]', {
				"Content-Type": "application/json",
			});

		await expect(
			createRecordVo(
				{
					bearerToken: "12345",
					baseUrl: "https://permanent.local/api",
				},
				{
					s3Url: "exampleUrl",
					displayName: "testFile",
					parentFolderId: 1,
					uploadFileName: "testFile.txt",
					fileType: "text/plain",
					size: 12,
				},
			),
		).rejects.toBeInstanceOf(ValidationError);
	});
});
