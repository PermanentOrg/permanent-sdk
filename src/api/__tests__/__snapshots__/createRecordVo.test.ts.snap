// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRecordVo should create a record 1`] = `
{
  "AccessVO": null,
  "ArchiveVOs": [],
  "FileVOs": null,
  "FolderLinkVOs": [
    {
      "accessRole": "access.role.owner",
      "archiveId": 2,
      "createdDT": "2023-01-23T21:02:40",
      "folderId": null,
      "folder_linkId": 14,
      "linkCount": 1,
      "parentFolderId": 7,
      "parentFolder_linkId": 7,
      "position": 1,
      "recordId": 5,
      "shareDT": null,
      "status": "status.generic.ok",
      "type": "type.folder_link.private",
      "updatedDT": "2023-01-23T21:02:40",
    },
  ],
  "LocnVO": null,
  "ParentFolderVOs": null,
  "RecordExifVO": null,
  "ShareVOs": [],
  "TagVOs": [],
  "TextDataVOs": [],
  "TimezoneVO": null,
  "accessRole": "access.role.owner",
  "archiveArchiveNbr": null,
  "archiveId": 2,
  "archiveNbr": "0002-000a",
  "createdDT": "2023-01-23T21:02:40",
  "derivedCreatedDT": null,
  "derivedDT": null,
  "derivedEndDT": null,
  "description": null,
  "displayDT": "2023-01-23T21:02:40",
  "displayEndDT": null,
  "displayName": "test",
  "downloadName": "test (1).txt",
  "downloadNameOk": true,
  "fileDurationInSecs": null,
  "fileStatus": "status.generic.undefined",
  "folderArchiveId": 2,
  "folder_linkId": 14,
  "folder_linkType": "type.folder_link.private",
  "imageRatio": 1,
  "locnId": null,
  "metaToken": "7JIO65WKIM",
  "note": null,
  "parentArchiveNbr": null,
  "parentDisplayName": null,
  "parentFolderId": 7,
  "parentFolder_linkId": 7,
  "pathAsArchiveNbr": null,
  "pathAsFolder_linkId": null,
  "pathAsText": null,
  "position": 1,
  "processedDT": null,
  "publicDT": null,
  "recordId": 5,
  "refArchiveNbr": null,
  "saveAs": null,
  "searchScore": null,
  "size": null,
  "status": "status.record.uploaded",
  "thumbDT": null,
  "thumbStatus": "status.record.needs_thumb",
  "thumbURL1000": null,
  "thumbURL200": null,
  "thumbURL2000": null,
  "thumbURL500": null,
  "timeZoneId": 88,
  "type": "type.record.document",
  "updatedDT": "2023-01-23T21:02:40",
  "uploadAccountId": 3,
  "uploadFileName": "test.txt",
  "uploadUri": null,
  "view": null,
  "viewProperty": null,
}
`;
